engine/src/main/scala/cromwell/engine/workflow/lifecycle/WorkflowExecutionActor.scala

```diff
@@ -221,6 +233,19 @@ final case class WorkflowExecutionActor(workflowId: WorkflowId, workflowDescript
       log.info(s"$tag transitioning from $fromState to $toState.")
   }
 
+  private def handleRetryableFailure(jobKey: BackendJobDescriptorKey) = {
```

Comment:
@gauravs90 This looks good to me, it's how it's working in `OldWorld` at least. I believe the code that's here now is not the one your first comment was referring to. I think you're right that strictly speaking we could overwrite the existing key in the store with NotStarted. The engine doesn't need to remember the failed attempts. The reason for the duplication was that we had a 1:1 mapping between the execution store and the Database, and the metadata came from the executions table.
Maybe keep it like this now, and we can remove it once Restart / Metadata is implemented and we're sure it's really useless.